Build failure: x3270

### Steps To Reproduce

Steps to reproduce the behavior:
1. build *x3270* on darwin fails due to missing macOs dependencies (libiconv/Security)

### Build log

```
error: builder for '/nix/store/mrfr4rsv34x5p1j2hg9k5x4azhjpbwwp-x3270-4.0ga9.drv' failed with exit code 1;
       last 10 log lines:
       > checking for forkpty... yes
       > checking for library containing gethostbyname... none required
       > checking for library containing socket... none required
       > checking iconv.h usability... yes
       > checking iconv.h presence... yes
       > checking for iconv.h... yes
       > checking for library containing libiconv... no
       > checking for library containing iconv... no
       > configure: error: No iconv library function
       > configure: error: ./configure failed for lib
```

### Additional context

The are a few things missing that are need to get this to work on macOS
* use & link iconv
* use Security package for headers
* disable x3270 build (the other unix builds are fine)

### Notify maintainers

@anna328p 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
→ nix-shell -p nix-info --run "nix-info -m"
these 2 paths will be fetched (0.01 MiB download, 0.11 MiB unpacked):
  /nix/store/xkkpiw8wj0cglvkzpp8gd2h5mgsq1kik-DarwinTools-1
  /nix/store/highid3c54gk59sxhibjhkyq2s2yvrch-nix-info
copying path '/nix/store/xkkpiw8wj0cglvkzpp8gd2h5mgsq1kik-DarwinTools-1' from 'https://cache.nixos.org'...
copying path '/nix/store/highid3c54gk59sxhibjhkyq2s2yvrch-nix-info' from 'https://cache.nixos.org'...
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.2.0, macOS 14.2.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.17.0`
 - channels(scott): `"home-manager, nixpkgs"`
 - channels(root): `"home-manager, nixpkgs"`
 - nixpkgs: `/Users/scott/.nix-defexpr/channels/nixpkgs`
```
---

I have a fixed package for this that I can open a PR for

--- a/pkgs/applications/terminal-emulators/x3270/default.nix
+++ b/pkgs/applications/terminal-emulators/x3270/default.nix
@@ -1,39 +1,72 @@
-{ lib, stdenv, fetchurl, m4, expat
-, libX11, libXt, libXaw, libXmu, bdftopcf, mkfontdir
-, fontadobe100dpi, fontadobeutopia100dpi, fontbh100dpi
-, fontbhlucidatypewriter100dpi, fontbitstream100dpi
+{ stdenv
+, darwin
+, lib
+, libiconv
+, fetchurl
+, m4
+, expat
+, libX11
+, libXt
+, libXaw
+, libXmu
+, bdftopcf
+, mkfontdir
+, fontadobe100dpi
+, fontadobeutopia100dpi
+, fontbh100dpi
+, fontbhlucidatypewriter100dpi
+, fontbitstream100dpi
 , tcl
-, ncurses }:
-
+, ncurses
+}:
 let
   majorVersion = "4";
-  minorVersion = "0";
-  versionSuffix = "ga9";
-in stdenv.mkDerivation rec {
+  minorVersion = "3";
+  versionSuffix = "ga8";
+in
+stdenv.mkDerivation rec {
   pname = "x3270";
   version = "${majorVersion}.${minorVersion}${versionSuffix}";
 
   src = fetchurl {
-    url = "http://x3270.bgp.nu/download/0${majorVersion}.0${minorVersion}/suite3270-${version}-src.tgz";
-    sha256 = "0km24rgll0s4ji6iz8lvy5ra76ds162s95y33w5px6697cwqkp9j";
+    url =
+      "http://x3270.bgp.nu/download/0${majorVersion}.0${minorVersion}/suite3270-${version}-src.tgz";
+    sha256 = "sha256-gcC6REfZentIPEDhGznUSYu8mvVfpPeMz/Bks+N43Fk=";
   };
 
-  buildFlags = [ "unix" ];
+  buildFlags = lib.optional stdenv.isLinux "unix";
 
-  postConfigure = ''
-    pushd c3270 ; ./configure ; popd
+  configureFlags = lib.optionals stdenv.isDarwin [
+    "--enable-c3270"
+    "--enable-pr3270"
+    "--enable-s3270"
+    "--enable-tcl3270"
+  ];
+
+  postBuild = ''
+    make install.man
   '';
+
+  pathsToLink = [ "/share/man" ];
 
   nativeBuildInputs = [ m4 ];
   buildInputs = [
     expat
-    libX11 libXt libXaw libXmu bdftopcf mkfontdir
-    fontadobe100dpi fontadobeutopia100dpi fontbh100dpi
-    fontbhlucidatypewriter100dpi fontbitstream100dpi
+    libX11
+    libXt
+    libXaw
+    libXmu
+    bdftopcf
+    mkfontdir
+    fontadobe100dpi
+    fontadobeutopia100dpi
+    fontbh100dpi
+    fontbhlucidatypewriter100dpi
+    fontbitstream100dpi
     tcl
     ncurses
     expat
-  ];
+  ] ++ lib.optionals stdenv.isDarwin [ libiconv darwin.apple_sdk.frameworks.Security ];
 
   meta = with lib; {
     description = "IBM 3270 terminal emulator for the X Window System";

